submissions/asaMitaka/OOP exercise/indexClass.js

```diff
@@ -0,0 +1,70 @@
+class Mammal  {
+    constructor (name, gender, legs, saying) {
+        this.species = 'mammal';
+        this.name = name;
+        this.gender = gender;
+        this.legs = legs;
+        this.saying = saying;
+    }
+
+    tellAboutClass() {
+        return `${this.saying} My name is ${this.name}. I'm ${this.species} ${this.gender}. I have ${this.legs} legs.`;
+    }
+}
+
+class Human extends Mammal{
+    constructor (name, gender, saying) {
+        super(name, gender);
+        this.species = 'human';
+        this.legs = 2;
+        this.hands = 2;
+        this.saying = saying;
+    }
+
+    tellAboutClass() {
+        return `${super.tellAboutClass()} I have ${this.hands} hands.`;
+    }
+}
+
+class Man extends Human {
+    constructor (name, saying = 'Hello.') {
+        super(name, saying);
+        this.gender = 'male';
+        this.saying = saying;
+    }
+}
+
+class Woman extends Human {
+    constructor (name, saying = 'Hi') {
+        super(name, saying);
+        this.gender = 'female';
+        this.saying = saying;
```

Comment:
There are a lot of repeating actions, for example - `saying`. When you create `Woman`, you pass a string as a `saying` to it. It gets it as a constructor parameter and then passes it to the `super` constructor. Later, on the `41` line it assigns this value to the `this. saying` property. But what does `super` do with it? Well, here, basically there's broken logic since the second parameter of the `Human` constructor is gender. So you pass saying as gender. And you also rewrite `gender` on line 40. Do you see what I'm saying? Please restructure your code, maybe even from scratch to make sure that every class is responsible for its exclusive properties and if something needs to be set from outside (like names or genders) it is passed properly.